### PR TITLE
Header component

### PR DESCRIPTION
**Pull request by willbeaumont**
Issue: #42 

## Summary (1-2 sentences)

This PR simplifies dashboard development by abstracting styling for h1 header components. The new component is applied to the `groups/new` page.  Includes recommendations from closed PR.

## Details (reason and description of the changes)

Having a reusable component will make it easier to update dashboard header styles if they need to change in the future. We can now update in one location to update multiple pages. The style settings are the closest available default tailwindcss styles that match the figma design doc.

### Implementation details (how was this change implemented?) 

This change is dependent on @andrelandgraf's Implement loading indications (login, signup), improve layout (groups/new) PR.

### CODE REVIEW

**Comment by willbeaumont** [comment]
I wish I would have squashed my commits.

Should we limit the content to be string type? Would it be weird if people wrapped a component in the H1 /H2 tag?

File: app/components/ui/containers.tsx [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export function Card({ children }: { children: React.ReactNode }) {
   return <section className="w-full bg-white shadow-lg rounded-lg px-2 py-6 lg:p-16">{children}</section>;
 }
+
+export function DashboardHeader({ children }: { children: React.ReactNode }) {
```
Comment (andrelandgraf) [inline app/components/ui/containers.tsx:5]: You probably want to remove this again!

File: app/components/ui/headers.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export function H1({ children }: { children: React.ReactNode }) {
```
Comment (andrelandgraf) [inline app/components/ui/headers.tsx:1]: Looks great! Thanks for creating the file!

> Should we limit the content to be string type? Would it be weird if people wrapped a component in the H1 /H2 tag?

I like the `children` implementation that you currently have. It matches how the `h1`, `h2` HTML elements work and also allows us to nest `span`, `a` or other elements within the headings.

**Review by andrelandgraf — COMMENTED** [review]
Thanks for making the changes. 

If you merge the latest `main` branch into your feature branch and push the changes, then you should be all good!

```
git checkout main
git pull
git checkout willbeaumont:header-component
git merge main
git push
```

You will probably have to resolve the merge conflict where you changed `main` and use the new components, but should be doable!

File: app/components/ui/containers.tsx (hunk 1 above)
Comment (willbeaumont) [reply app/components/ui/containers.tsx:5]: Doh! 🙈

**Review by andrelandgraf — APPROVED** [review]
Awesome! Let's get it merged!

**Review by shami-dev — APPROVED** [review]

**Pull request merged**